src/coreclr/vm/object.h

```diff
@@ -1274,6 +1274,12 @@ class ThreadBaseObject : public Object
         return m_Name;
     }
 
+    OBJECTREF GetExecutionContext()
+    {
+        LIMITED_METHOD_CONTRACT;
+        return m_ExecutionContext;
+    }
```

Comment:
Is this being used?

(Note that with #122125 there are cases where we cannot permit direct use)

Reply:
Yes. We explicitly don't use the AsyncHelpers helper for this as it adds yet another instruction to the sequence needed, and each one of those is pretty expensive. I'll adjust the logic in interpexec to do the right thing, but #122125 really needs to include a targeted testcase so that I can validate the behavior.

Reply:
I'll add a few test scenarios with https://github.com/dotnet/runtime/pull/122125

There are Libraries tests sensitive to the suppression of execution context flow, but they are tests for Unix Sockets. 
Perhaps not a good place to be in terms of test coverage.